src/data/real/ereal.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2019 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard
+-/
+
+import data.real.basic 
+
+/-!
+# The extended reals [-∞, ∞].
+
+This file defines `ereal`, the real numbers together with a top and bottom element,
+referred to as ⊤ and ⊥. 
+
+Addition and multiplication are problematic in the presence of ±∞, but
+negation is not, so we define negation. The main work is defining
+Sup : set ereal → ereal and proving that it is a Sup in all cases.
+Once we have, this, it is not hard to give a complete lattice
+structure on `ereal`.
+
+## Tags
+
+real, ereal, complete lattice
+-/
+
+/-- ereal : The type $$[-\infty,+\infty]$$ or `[-∞, ∞]` -/
+def ereal := with_bot (with_top ℝ)
+
+instance : linear_order ereal := by unfold ereal; apply_instance
```

Comment:
Did you try the [new gadget](https://github.com/leanprover-community/mathlib/blob/dfd25ff4fe93bbe6e376dd2d27f7ef7ee7c8b3f9/src/tactic/core.lean#L1363-L1375) deriving instances?

Reply:
Thanks! I didn't know about this.